dev/EnvironmentManager/API/Microsoft.Windows.System.EnvironmentManager.Insights.h

```diff
@@ -0,0 +1,45 @@
+﻿#pragma once
+#include "Microsoft.Windows.System.EnvironmentManager.h"
+
+class EnvironmentManagerInsights : public wil::TraceLoggingProvider
+
+{
+    IMPLEMENT_TRACELOGGING_CLASS(
+        EnvironmentManagerInsights,
+        "Microsoft.Windows.System.EnvironmentManager.Insights",
+        // {fbce41fa-f7bc-538e-11f2-abe1b9d2ab23}
+        (0xfbce41fa, 0xf7bc, 0x538e, 0x11, 0xf2, 0xab, 0xe1, 0xb9, 0xd2, 0xab, 0x23));
+
+    DEFINE_COMPLIANT_MEASURES_EVENT_STRING(StringTelemetryEvent, // Event name 
+        MICROSOFT_KEYWORD_MEASURES, // Check wil/traceloggingconfig.h for a full list 
+        stringvalue/* value to be logged*/);
+
+    DEFINE_COMPLIANT_TELEMETRY_EVENT_PARAM2(ScopeTelemetryEvent, //Event name
+        MICROSOFT_KEYWORD_MEASURES, // Privacy data data for events. Check wil/traceloggingconfig.h for a full list
+        winrt::Microsoft::Windows::System::implementation::EnvironmentManager::Scope, // Payload data type
+        value,
+        PCWSTR, message);
+
+    static void LogWithScopeAndMessage(winrt::Microsoft::Windows::System::implementation::EnvironmentManager::Scope const& scope,
+        PCWSTR message)
+    {
+        std::wstring scopeAsString{};
+
+        if (scope == winrt::Microsoft::Windows::System::implementation::EnvironmentManager::Scope::Process)
+        {
+            scopeAsString = L"Process";
+        }
+        else if (scope == winrt::Microsoft::Windows::System::implementation::EnvironmentManager::Scope::User)
+        {
+            scopeAsString = L"User";
+        }
+        else
+        {
+            scopeAsString = L"Machine";
+        }
+
+        TraceLoggingClassWrite("CustomEvent", TraceLoggingValue(scopeAsString.c_str(), "scope"),
```

Comment:
>TraceLoggingClassWrite [](http://example.com/codeflow?start=8&length=22)

This will not log the PartB information. Any reason why you are calling TraceLoggingClassWrite explicitly instead of defining an event (CustomEvent) and logging the data as payload?

Reply:
![image](https://user-images.githubusercontent.com/2517139/133826148-ae3afabd-e726-4d9b-b74f-ac49324bb40d.png)

By partB i'm guessing you mean the message?  Both are logged.

I'm calling TraceLoggingClassWrite because that is how I saw how to call a custom event.  Is there a better way to call a custom event?

I could convert the scope to a string and add it to a std::wstring and call StringTelemetryEvent and that way there isn't a need for a customer event.

Is there a different way to call a custom event?

Reply:
Nope. PartB contains the information about the SDK and other fields that must be logged with all data coming from WindowsAppSDK. This will help us produce metrics on the usage and stability of code in different versions of WindowsAppSDK.

Reply:
You could have a member similar to m_scope like m_scopeString and log that as part of the payload in the event. Alternately, you could log the enum (as int) as part of the payload.